proposals/apple-mdm/Fleet-MDM-Setup.md

```diff
@@ -0,0 +1,127 @@
+# Setup
+
+The setup consists of configuring:
+- The APNs certificate used by the MDM protocol
+- The SCEP certificate for enrollment
+
+We will define `fleetctl apple-mdm setup ...` commands to create/define all Apple/MDM credentials that are fed to the Fleet server.
+
+## APNs
+
+### Apple MDM APNs setup
+
+Apple's MDM protocol uses the Apple Push Notification service (APNs) to deliver "wake up" messages to managed devices.
+An "MDM server" needs access to an APNS certificate specifically issued for MDM management; such APNs certificate must be issued by an "MDM vendor."
+
+Here's a sequence diagram with the three actors: Apple Inc., an MDM vendor, and a customer (MDM server).
+
+```mermaid
+%%{init: { 'theme':'dark', 'sequence': {'mirrorActors':false} } }%%
+
+sequenceDiagram
+    participant apple as Apple Inc.
+    participant vendor as MDM Vendor
+    participant server as MDM Server<br>(Customer)
+
+    rect rgb(128, 128, 128)
+    note left of apple: (1) MDM Vendor Setup at<br>https://business.apple.com
+    note over vendor: Generate CSR
+    vendor->>apple: Send CSR
+    note over apple: Sign CSR
+    apple->>vendor: "MDM vendor" certificate<br>(Setup)
+    end
+    rect rgb(255, 255, 255)
+    note left of apple: (2) Customer Setup
+    note over server: Generate CSR
+    server->>+vendor: "Send" CSR (.csr)
+    note over vendor: Sign CSR
+    vendor->>server: "Send" signed CSR (XML plist, .req)
+    rect rgb(128, 128, 128)
+    note left of apple: https://identity.apple.com/pushcert
+    server->>apple: Upload signed CSR (XML plist, .req)
+    note over apple: Sign CSR
+    apple->>server: Download APNS Certificate (PEM)
+    end
+    end
+    note over server: APNS keypair<br>ready to use
+```
+
+The "MDM vendor setup" flow (1) is executed once by the "MDM vendor."
+
+The "customer setup" flow (2) is executed by customers when they are setting up their MDM server.
+
+The goal is for the Fleet organization to become an "MDM vendor" that issues CSRs to customers, which allows them to generate "APNs certificates" for their MDM deployments.
+
+For the purposes of designing a PoC, we used the https://mdmcert.download/ service as an "MDM vendor."
+See [MDMCert.Download Analysis](./mdmcert.download-analysis.md) for more details on the process.
+
+### APNs setup with Fleet
+
+The MDM APNs certificate provisioning will be manual on MVP:
+- Customers will use `fleetctl` commands that will mimick `mdmctl mdmcert.download` commands (see [MDMCert.Download Analysis](mdmcert.download-analysis.md)).
+- Fleet operators will perform the steps shown in the diagram above manually by running a new command line tool (under `tools/mdm-apple/mdm-apple-customer-setup`).
+
+#### 1. Init APNs (customer)
+
+`fleetctl apple-mdm setup apns init` 
```

Comment:
Is there anything else to setup besides `apns`? If so, can we drop the apns namespace?

Reply:
Yes, there will be a `setup scep` too (to setup the CA in Fleet, for [SCEP](https://en.wikipedia.org/wiki/Simple_Certificate_Enrollment_Protocol), needed for implementing MDM).